src/liballoc/rc.rs

```diff
@@ -591,17 +591,13 @@ impl<T: ?Sized> Rc<T> {
     #[stable(feature = "weak_into_raw", since = "1.45.0")]
     pub fn as_ptr(this: &Self) -> *const T {
         let ptr: *mut RcBox<T> = NonNull::as_ptr(this.ptr);
-        let fake_ptr = ptr as *mut T;
 
         // SAFETY: This cannot go through Deref::deref.
         // Instead, we manually offset the pointer rather than manifesting a reference.
```

Comment:
"manually offset the pointer" doesn't apply any more.

But the comment needs to argue why `ptr` is dereferencable and aligned.

Reply:
(Partially?) addressed. This is `Rc::as_ptr`, so the pointer is always valid here. I think noting that this is doing the same thing as the safe function `Deref::deref` is probably enough to remind the reader that this is in the `Rc` context where the pointer is guaranteed valid.